helm/spark-yotpo/templates/sparkapplication.yaml

```diff
@@ -0,0 +1,161 @@
+  mainClass: {{ .Values.mainClass | quote }}
+  mainApplicationFile: {{ .Values.jar | quote }}
+  {{- if .Values.arguments }}
+  arguments:
+  {{- range .Values.arguments }}
+    - {{ . | quote }}
+  {{- end }}
+  {{- end }}
+  sparkVersion: {{ .Values.sparkVersion | quote }}
+  {{- if or .Values.hiveMetaStore.enabled .Values.builtinMetaStore.enabled .Values.atlas.enabled }}
+  sparkConf:
+  {{- if .Values.hiveMetaStore.enabled  }}
+  {{- range $conf, $value := .Values.hiveMetaStore.conf }}
+    {{ $conf | quote }}: {{ $value | quote }}
+  {{- end }}
+  {{- end }}
+  {{- if .Values.builtinMetaStore.enabled  }}
+  {{- range $conf, $value := .Values.builtinMetaStore.conf }}
+    {{ $conf | quote }}: {{ $value | quote }}
+  {{- end }}
+  {{- end }}
+  {{- if .Values.atlas.enabled  }}
+  {{- range $conf, $value := .Values.atlas.conf }}
+    {{ $conf | quote }}: {{ $value | quote }}
+  {{- end }}
+  {{- end }}
+  {{- end }}
+  restartPolicy:
+    type: Never
+  {{- if or .Values.jarDependencies .Values.fileDependencies .Values.sparkConf .Values.hadoopConf }}
+  deps:
+    {{- if .Values.jarDependencies }}
+    jars:
+    {{- range .Values.jarDependencies }}
+      - {{ . | quote }}
+    {{- end }}
+    {{- end }}
+    {{- if .Values.fileDependencies }}
+    files:
+    {{- range .Values.fileDependencies }}
+      - {{ . | quote }}
+    {{- end }}
+    {{- end }}
+    {{- if .Values.hadoopConf }}
+    hadoopConf:
+    {{- range $conf, $value := .Values.hadoopConf }}
+      {{ $conf | quote }}: {{ $value | quote }}
+    {{- end }}
+    {{- end }}
+  {{- end }}
+  driver:
+    {{- if .Values.envSecretKeyRefs }}
+    envSecretKeyRefs:
+    {{- range $name, $value := .Values.envSecretKeyRefs }}
+      {{ $name }}:
+        name: {{ $value.name}}
+        key: {{ $value.key}}
+    {{- end }}
+    {{- end }}
+    {{- if .Values.envVars }}
+    envVars:
+    {{- range $name, $value := .Values.envVars }}
+      {{ $name }}: {{ $value | quote }}
+    {{- end }}
+    {{- end }}
+    securityContext:
+      runAsUser: {{ .Values.userId }}
+    cores: {{ .Values.driver.cores }}
+    coreLimit: {{ .Values.driver.coreLimit | default .Values.driver.cores | quote }}
+    memory: {{ .Values.driver.memory }}
+    hostNetwork: {{ .Values.hostNetwork }}
+    labels:
+      release: {{ .Release.Name | trunc 63 | quote }}
+      revision: {{ .Release.Revision | quote }}
+      sparkVersion: {{ .Values.sparkVersion | quote }}
+      version: {{ .Chart.Version | quote }}
+    serviceAccount: {{ .Values.serviceAccount }}
+    {{- if .Values.javaOptions }}
+    javaOptions: {{ .Values.javaOptions | quote}}
+    {{- end }}
+    {{- if .Values.driver.mounts }}
+    volumeMounts:
+    {{- range $name, $path := .Values.driver.mounts }}
+      - name: {{ $name }}
+        mountPath: {{ $path }}
+    {{- end }}
+    {{- end }}
+    {{- if .Values.tolerations }}
+    tolerations:
+{{ toYaml .Values.tolerations | indent 6 }}
+    {{- end }}
+  executor:
+    {{- if .Values.envVars }}
```

Comment:
I don't know if spark needs this or not, but it looks like you define envVars only once
(.Values.envVars) and it applies to both executor and driver. this means you can't have seperate vars for both. 
Just FYI, not sure if it's by design or not.